lib/ansible/modules/network/exos/exos_guest_virtual_machines.py

```diff
@@ -0,0 +1,1222 @@
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The module file for exos_guest_virtual_machines
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = """
+---
+module: exos_guest_virtual_machines
+version_added: "2.10"
+short_description:  Manage guest virtual machines on EXOS devices.
+description:
+  - This module can be used to create guest Virtual machines on EXOS network devices
+  - Manages operational state such as start, stop and restart
+  - Updates its components such as CPUs, memory, virtual ports, disks etc.
+  - Deletes a guest virtual machine with all its associated components.
+author: Jayalakshmi Viswanathan (@jayalakshmiV)
+notes:
+  - Tested against EXOS 30.5
+  - This module works with connection C(httpapi).
+    See L(EXOS Platform Options,../network/user_guide/platform_exos.html)
+options:
+  config:
+    description: A dictionary of Virtual machine configuration options
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description:
+          - Name of the virtual machine
+        type: str
+      image:
+        description:
+          - Name and location of the image on the switch to be installed
+        type: str
+      num_cores:
+        description:
+          - Number of CPU cores assigned to the virtual machine
+        type: int
+      memory_size:
+        description:
+          - Memory size in Megabytes assigned to the virtual machine
+        type: int
+      virtual_ports:
+        description:
+          - Virtual port to be used by the virtual machine.
+          - Only local Insight ports can be assigned as a dedicated port (type - vtd) or a virtual interface (type - sriov).
+          - An Insight port assigned as a dedicated port to a VM cannot be used to define a virtual interface for any VM or vice versa.
+          - An Insight port can have maximum of 16 active virtual interfaces associated with it at a time.
+        type: dict
+        suboptions:
+          name:
+            description:
+              - Name of the virtual port interface
+            type: str
+          port:
+            description:
+              - Port associated with the virtual port
+            type: str
+          vlan:
+            description:
+              - Vlan to which the virtual port is to be attached
+            type: int
+          type:
+            description:
+              - Virtual port connect type.
+            type: str
+            choices:
+              - vtd
+              - sriov
+              - bridge
+      vnc:
+        description:
+          - VNC configuration for the virtual machine.
+        type: dict
+        suboptions:
+          enabled:
```

Comment:
Would this be a required field?

Reply:
It's not a required field. It's by default 'false'.